### PR TITLE
WIP: feat(admin-setting): improve success/failed/history page settings #2445

### LINKED ISSUE #2445: feat(admin-setting): improve success/failed/history page settings
## User story
As an admin, I want Give pages setting to more flexible to set Give specific pages easily. Currently, I can only see a limited number of pages.

## Issue Overview

image

## Tasks.
- [x] Add apply chosen js for each pages setting.
- [x] By default show 30 page in list.
- [x] Implement search page by ajax.

### PR DESCRIPTION
**Pull request by Sidsector9**
Closes #2445 

## Description
Added AJAX-powered chosen.js to search pages easily

## How Has This Been Tested?
This has been manually tested.

## Screenshots (jpeg or gifs if applicable):
chosenjs

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/ajax-functions.php [hunk 1]
```diff
@@ -586,4 +586,39 @@ function give_confirm_email_for_donation_access() {
...
+ * @since 2.1
+ *
+ * @return json
+ */
+function give_ajax_pages_search() {
+
+	$data = [];
+	$args = array(
+		'post_type' => 'page',
+		's'         => $_GET['s'],
```
Comment (ravinderk) [inline includes/ajax-functions.php]: @Sidsector9 You must escape query params.

File: includes/ajax-functions.php [hunk 2]
```diff
@@ -586,4 +586,39 @@ function give_confirm_email_for_donation_access() {
...
-add_action( 'wp_ajax_nopriv_give_confirm_email_for_donations_access', 'give_confirm_email_for_donation_access' );
\ No newline at end of file
+add_action( 'wp_ajax_nopriv_give_confirm_email_for_donations_access', 'give_confirm_email_for_donation_access' );
+
+/**
+ * Queries page by title and returns page ID and title in JSON format.
+ *
+ * @since 2.1
+ *
+ * @return json
```
Comment (ravinderk) [inline includes/ajax-functions.php]: @Sidsector9 result type will be `string` here

File: includes/ajax-functions.php [hunk 3]
```diff
@@ -586,4 +586,39 @@ function give_confirm_email_for_donation_access() {
...
+			$query->the_post();
+
+			$data[] = array(
+				'id' => get_the_ID(),
+				'name' => get_the_title(),
+			);
+		}
+	}
+
+	echo wp_json_encode( $data );
```
Comment (ravinderk) [inline includes/ajax-functions.php]: @Sidsector9 Use `wp_send_json` instead

**Comment by ravinderk** [comment]
@Sidsector9 Can you test this PR?

**Comment by Sidsector9** [comment]
@ravinderk The results that you get after search are not being saved.
Reproduce issue:
1. Limit `numberposts` to `3`
2. Search for a page which is not in the list of 3 pages
3. Try saving the searched page (It won't save)

**Comment by Sidsector9** [comment]
@ravinderk I have tested this, it works. :+1:

**Pull request merged**